whois-api/src/main/java/net/ripe/db/whois/api/httpserver/RemoteAddressCustomizer.java

```diff
@@ -43,4 +60,12 @@ private String getRemoteAddress(final Request request) {
 
         return Iterables.getLast(COMMA_SPLITTER.split(header));
     }
+
+    private Set<Interval> getIntervals(final String[] trusted) {
+        return Arrays.stream(trusted).map(IpInterval::parse).collect(Collectors.toSet());
+    }
+
+    private boolean isTrusted(final Interval ipResource, final Set<Interval> ipRanges) {
+        return ipRanges.stream().anyMatch(ipRange -> ipRange.getClass().equals(ipResource.getClass()) && ipRange.contains(ipResource));
```

Comment:
Do we need to review the class?
If one of them is an Ipv4 and the other is an ipv6 the contains will take care of it anyways.
and instead of the contains is should be equals right? in the end you are comparing an Interval with another Interval. You are not checking if the interval exists in one list

Reply:
Yes as its two seperate class for ipv4 and ipv6 else it can throw class cast exception